### ISSUE TITLE
Syntax error in lib/listeners/http_hop.generate_launcher() [python]

### ISSUE THREAD

**Issue by johneiser** [EmpireProject/Empire#1053]
## Empire Version

2.5

## OS Information (Linux flavor, Python version)

[Server] Ubuntu 16.04 x64, Python 2.7.12
[Server] Kali 4.14 x64, Python 2.7.14+
[Client] macOS High Sierra 10.13.3, Python 2.7.10

## Expected behavior and description of the error, including any actions taken immediately prior to the error. The more detail the better.

lib/listeners/http_hop.generate_launcher() redefines the variable 'o' on line 295, which overwrites the definition on line 264 and erases the UA and cookie headers added on line 273.  Recommend moving line 273 to 296.

Issue:
```
launcherBase += "o=__import__({2:'urllib2',3:'urllib.request'}[sys.version_info[0]],fromlist=['build_opener']).build_opener();"
...
# add the RC4 packet to a cookie
launcherBase += "o.addheaders=[('User-Agent',UA), (\"Cookie\", \"session=%s\")];\n" % (b64RoutingPacket)
launcherBase += "import urllib2\n"
...
else:
    launcherBase += "o = urllib2.build_opener();\n"
```

Solution:
```
launcherBase += "o=__import__({2:'urllib2',3:'urllib.request'}[sys.version_info[0]],fromlist=['build_opener']).build_opener();"
...
launcherBase += "import urllib2\n"
...
else:
    launcherBase += "o = urllib2.build_opener();\n"

# add the RC4 packet to a cookie
launcherBase += "o.addheaders=[('User-Agent',UA), (\"Cookie\", \"session=%s\")];\n" % (b64RoutingPacket)
```

## Screenshot of error, embedded text output, or Pastebin link to the error

Setup:
```
(Empire: stager/osx/macho) > info

Name: macho

Description:
  Generates a macho executable.

Options:

  Name             Required    Value             Description
  ----             --------    -------           -----------
  Listener         True        http_hop          Listener to generate stager for.
  OutFile          True        /var/www/html/stagerFile to write macho to.
  SafeChecks       True        False             Switch. Checks for LittleSnitch or a
                                                 SandBox, exit the staging process if
                                                 true. Defaults to True.
  Language         True        python            Language of the stager to generate.
  UserAgent        False       default           User-agent string to use for the staging
                                                 request (default, none, or other).

(Empire: listeners) > info http_hop

http_hop Options:

  Name              Required    Value                            Description
  ----              --------    -------                          -----------
  RedirectListener  True        http                             Existing listener to redirect the hop traffic to.
  Name              True        http_hop                         Name for the listener.
  Launcher          True        powershell -noP -sta -w 1 -enc   Launcher string.
  RedirectStagingKeyFalse       IyG@n=(7B_kbHlX8m~}aEO+VfY^)Q/-. The staging key for the redirect listener, extracted from RedirectListener automatically.
  SlackToken        False                                        Your SlackBot API token to communicate with your Slack instance.
  Host              True        http://192.168.3.113:80          Hostname/IP for staging.
  SlackChannel      False       #general                         The Slack channel or DM that notifications will be sent to.
  OutFolder         True        /var/www/html                    Folder to output redirectors to.
  DefaultProfile    False       /admin/get.php,/news.php,/login/ Default communication profile for the agent, extracted from RedirectListener automatically.
                                process.php|Mozilla/5.0 (Windows
                                NT 6.1; WOW64; Trident/7.0;
                                rv:11.0) like Gecko
  Port              True        80                               Port for the listener.


(Empire: listeners) > info http

http Options:

  Name              Required    Value                            Description
  ----              --------    -------                          -----------
  StagerURI         False                                        URI for the stager. Must use /download/. Example: /download/stager.php
  ProxyCreds        False       default                          Proxy credentials ([domain\]username:password) to use for request (default, none, or other).
  KillDate          False                                        Date for the listener to exit (MM/dd/yyyy).
  Name              True        http                             Name for the listener.
  Launcher          True        powershell -noP -sta -w 1 -enc   Launcher string.
  DefaultProfile    True        /admin/get.php,/news.php,/login/ Default communication profile for the agent.
                                process.php|Mozilla/5.0 (Windows
                                NT 6.1; WOW64; Trident/7.0;
                                rv:11.0) like Gecko
  DefaultLostLimit  True        60                               Number of missed checkins before exiting
  Host              True        http://192.168.3.113:9999        Hostname/IP for staging.
  Port              True        9999                             Port for the listener.
  WorkingHours      False                                        Hours for the agent to operate (09:00-17:00).
  CertPath          False                                        Certificate path for https listeners.
  DefaultJitter     True        0.0                              Jitter in agent reachback interval (0.0-1.0).
  SlackChannel      False       #general                         The Slack channel or DM that notifications will be sent to.
  BindIP            True        0.0.0.0                          The IP to bind to on the control server.
  UserAgent         False       default                          User-agent string to use for the staging request (default, none, or other).
  StagingKey        True        IyG@n=(7B_kbHlX8m~}aEO+VfY^)Q/-. Staging key for initial agent negotiation.
  DefaultDelay      True        5                                Agent delay/reach back interval (in seconds).
  SlackToken        False                                        Your SlackBot API token to communicate with your Slack instance.
  ServerVersion     True        Microsoft-IIS/7.5                Server header for the control server.
  Proxy             False       default                          Proxy to use for request (default, none, or other).


```

Agent (osx/macho) Error:
```
$ ./stager
Traceback (most recent call last):
  File "<string>", line 16, in <module>
TypeError: expected string without null bytes
```

Traffic:
```
GET /login/process.php HTTP/1.1
Accept-Encoding: identity
Host: 192.168.3.113:80
Connection: close
User-Agent: Python-urllib/2.7

HTTP/1.1 200 OK
Date: Fri, 30 Mar 2018 00:59:31 GMT
Server: Apache/2.4.18 (Ubuntu)
Vary: Accept-Encoding
Content-Length: 1217
Connection: close
Content-Type: text/html; charset=UTF-8

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
```

Empire Error:
```
[!] login/process.php requested by 192.168.3.113 with no routing packet
```

**Comment by xorrior** [comment]
@johneiser I'll take a look at this

**Comment by xorrior** [comment]
Fixed in #1062
